components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/inbound/websocket/InboundWebSocketProcessor.java

```diff
@@ -156,7 +156,11 @@ public InboundProcessorResponseDTO handleHandshake(FullHttpRequest req, ChannelH
             log.error(e.getMessage());
             inboundProcessorResponseDTO = InboundWebsocketProcessorUtil.getHandshakeErrorDTO(
                     WebSocketApiConstants.HandshakeErrorConstants.RESOURCE_NOT_FOUND_ERROR, e.getMessage());
-            publishResourceNotFoundEvent(ctx);
+
+            boolean isResourceNotFoundErrorCodeExist = WebSocketUtils.getApiProperties(ctx).containsKey(SynapseConstants.ERROR_CODE);
+            if (isResourceNotFoundErrorCodeExist) {
```

Comment:
Let's remove the extra boolean variable, and do like:
```java
if (WebSocketUtils.getApiProperties(ctx).containsKey(SynapseConstants.ERROR_CODE)) {
```

Reply:
Resolved a2361854efc46b5a163bf8ebb55b3b19eb7c41c2